src/cryptoApprove/impl/api/index.ts

```diff
@@ -591,11 +617,9 @@ export class ApiCryptoApproveProvider implements ICryptoApproveProvider {
     const chainId = crypto.chainId;
     const actorAddress = owner;
     const tokenAddress = crypto.address;
-    const ownerWallet = await wallet.getOwnerWallet();
-    const ownerAddress = await ownerWallet.getAddress();
+    const smartWalletAddress = await wallet.getAddress({ chainId });
 
-    const { encodeErc20Approve } = await import('./erc20');
-    const tokenApproveData = await encodeErc20Approve(spender, amount);
+    const data = await this.makeSmartApproveData(spender, amount, crypto);
```

Comment:
does this work with possible `pre` below?

in short - sometimes we need two transactions executed in batch: `approve(0) + approve(!0)`

because some tokens don't allow `!0 -> !0`, only `!0 -> 0 -> !0`